hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/BaseActionExecutor.java

```diff
@@ -157,7 +157,7 @@ protected final void writeTableMetadata(HoodieRestoreMetadata metadata) {
    */
   protected final void dropIndexOnRestore() {
     for (String partitionPath : table.getMetaClient().getTableConfig().getMetadataPartitions()) {
-      if (MetadataPartitionType.shouldDeletePartitionOnRestore(partitionPath)) {
+      if (MetadataPartitionType.shouldDeleteMDTPartitionOnOverwriteOrRestore(partitionPath)) {
```

Comment:
lets not re-create the index automatically. its ok to throw exception and suggest users to disable SI for the table of interest. 
they can wait for 1.0.1 or something to get a fix for this.

Reply:
Addressed